components/http-accept.js

```diff
@@ -1,55 +1,49 @@
 
 var _ = require('underscore');
-var body = require('body');
-var noflo = require('noflo');
+var getRawBody = require('raw-body');
+var typer = require('media-typer');
 
 var basenode = require('./base-node');
+var promiseComponent = require('./promise-component');
 
-exports.getComponent = function() {
-    return _.extend(new noflo.Component({
-        outPorts: {
-            out: {
-                description: "Request body as a String",
-                datatype: 'string'
-            },
-            rejected: {
-                description: "HTTP 415 Unsupported Media Type request/response pair {req, res}",
-                datatype: 'object'
-            },
-            accepted: {
-                description: "HTTP 202 Accepted request/response pair {req, res}",
-                datatype: 'object'
-            }
+exports.getComponent = promiseComponent({
+    description: "Extracts the request body as a String and produces a 202 Accepted response. Optionally, also filters on request content type.",
+    icon: 'sign-in',
+    outPorts: {
+        rejected: {
+            description: "HTTP 415 Unsupported Media Type request/response pair {req, res}",
+            datatype: 'object'
         },
-        inPorts: {
-            limit: {
-                description: "Request Content-Length limit",
-                datatype: 'int',
-                process: basenode.on({data: basenode.assign('limit')})
-            },
-            encoding: {
-                description: "Request body character encoding",
-                datatype: 'string',
-                process: basenode.on({data: basenode.assign('encoding')})
-            },
-            type: {
-                description: "Request Content-Type",
-                datatype: 'string',
-                process: basenode.on({data: basenode.assign('types', basenode.push)})
-            },
-            'in': {
-                description: "HTTP request/response pair {req, res}",
-                datatype: 'object',
-                required: true,
-                process: basenode.on({data: handle})
-            }
+        accepted: {
+            description: "HTTP 202 Accepted request/response pair {req, res}",
+            datatype: 'object'
         }
-    }), {
-        description: "Extracts the request body as a String and produces a 202 Accepted response. Optionally, also filters on request content type.",
-        icon: 'sign-in'
-    });
-};
+    },
+    inPorts: {
+        limit: {
+            description: "Request Content-Length limit",
+            datatype: 'int',
+            ondata: basenode.assign('limit')
+        },
+        encoding: {
+            description: "Request body character encoding",
+            datatype: 'string',
+            ondata: basenode.assign('encoding')
+        },
+        type: {
+            description: "Request Content-Type",
+            datatype: 'string',
+            ondata: basenode.assign('types', basenode.push)
```

Comment:
When we get to it, I think the "ondata" property should be removed from all of the port definitions here, because (except for special components) all ports should automatically use the same function for handling data events.  However, we don't have to make that change in this PR.  It can be later.